python/pyspark/sql/types.py

```diff
@@ -1043,6 +1057,14 @@ def _infer_type(obj):
             raise TypeError("not supported type: %s" % type(obj))
 
 
+def _type_from_simple_string(simple_str):
+    """Map simpleString to Spark SQL type
+    """
+    found_type = _simple_type_map.get(simple_str, None)
```

Comment:
How about the more Pythonic
```
try: 
    found_type = _simple_type_map[simple_str]
except KeyError:
    raise TypeError("Unsupported type: %s" % simple_str)
```
Although I'm wondering if simply having the KeyError surfacing isn't enough.

Reply:
Updated 👍